Tests fail under alpha

## Description
<!--
If your issue is a bugreport, this means describing what you did,
what did you want to happen and what actually did happen.

If your issue is a feature request, describe the feature and why do you
want it.
-->

I'm attempting to validate this package on alpha for Gentoo, but I'm getting (inconsistent) failures in the test suite related to the timeout decorator.  Usually somewhere between 5 and 15 of the test cases will fail, the failures all look like this:

```
35/155 Testing: test_cases_and_suites.cpp_xml
35/155 Test: test_cases_and_suites.cpp_xml
Command: "/usr/bin/cmake" "-DCOMMAND=/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9_build/examples/all_features/all_features --no-skipped-summary --no-version -sf=*test_cases_and_suites.cpp --reporters=xml --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1" "-DTEST_OUTPUT_FILE=/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9/examples/all_features/test_output/test_cases_and_suites.cpp_xml.txt" "-DTEST_TEMP_FILE=/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9_build/examples/all_features/CMakeFiles/temp_test_output_test_cases_and_suites.cpp_xml.txt" "-DTEST_MODE=COMPARE" "-P" "/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9/scripts/cmake/exec_test.cmake"
Directory: /var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9_build/examples/all_features
"test_cases_and_suites.cpp_xml" start time: Oct 03 04:09 UTC
Output:
----------------------------------------------------------
-- Found Git: /usr/bin/git (found version "2.37.3") 
diff --git a/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9/examples/all_features/test_output/test_cases_and_suites.cpp_xml.txt b/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9_build/examples/all_features/CMakeFiles/temp_test_output_test_cases_and_suites.cpp_xml.txt
index 6fc245a..e9d1e79 100644
--- a/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9/examples/all_features/test_output/test_cases_and_suites.cpp_xml.txt
+++ b/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9_build/examples/all_features/CMakeFiles/temp_test_output_test_cases_and_suites.cpp_xml.txt
@@ -54,7 +54,7 @@
           failing because of the timeout decorator!
         </Text>
       </Message>
-      <OverallResultsAsserts successes="0" failures="0" test_case_success="false"/>
+      <OverallResultsAsserts successes="0" failures="0" test_case_success="true"/>
     </TestCase>
   </TestSuite>
   <TestSuite name="skipped test cases">
@@ -109,6 +109,6 @@
     </TestCase>
   </TestSuite>
   <OverallResultsAsserts successes="1" failures="11"/>
-  <OverallResultsTestCases successes="6" failures="9"/>
+  <OverallResultsTestCases successes="7" failures="8"/>
 </doctest>
 Program code.

CMake Error at /var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9/scripts/cmake/exec_test.cmake:69 (message):
  Running
  '/var/tmp/portage/dev-cpp/doctest-2.4.9/work/doctest-2.4.9_build/examples/all_features/all_features
  --no-skipped-summary --no-version -sf=*test_cases_and_suites.cpp
  --reporters=xml --dt-no-path-filenames=1 --dt-no-line-numbers=1
  --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1' ended
  with code 'Output is different from reference file!'

<end of output>
Test time =   0.22 sec
----------------------------------------------------------
Test Failed.
"test_cases_and_suites.cpp_xml" end time: Oct 03 04:09 UTC
"test_cases_and_suites.cpp_xml" time elapsed: 00:00:00
```

### Steps to reproduce
<!--
This is only relevant for bug reports, but if you do have one,
please provide a minimal set of steps to reproduce the problem.

Usually this means providing a small and self-contained code using doctest
and specifying compiler flags/tools used if relevant.
-->
```
mkdir build
cd build
cmake -G Ninja -DDOCTEST_WITH_TESTS=yes ..
ninja -v
ctest
```

### Extra information
<!--
Fill in any extra information that might be important for your issue.

If your issue is a bugreport, definitely fill out at least the following.
-->
* doctest version: **v2.4.9** and **git master** (tried both)
* Operating System: **Gentoo Linux - kernel 6.0**
* Compiler+version: **gcc 12.2.0**

Downstream bug, along with a complete sample log, is [here](https://bugs.gentoo.org/874345)

I have real hardware available that can be used to debug if necessary!  Let me know if this would be useful and I can provide access.

> Does the current version do the trick for you? I've moved to `std::chrono` but am not currently using `sleep_for` as that is causing issues with older versions of clang for x86..

Unfortunately still nope.

```
34/155 Testing: test_cases_and_suites.cpp
34/155 Test: test_cases_and_suites.cpp
Command: "/usr/bin/cmake" "-DCOMMAND=/tmp/doctest/build/examples/all_features/all_features --no-skipped-summary --no-version -sf=*test_cases_and_suites.cpp --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1" "-DTEST_OUTPUT_FILE=/tmp/doctest/examples/all_features/test_output/test_cases_and_suites.cpp.txt" "-DTEST_TEMP_FILE=/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_test_cases_and_suites.cpp.txt" "-DTEST_MODE=COMPARE" "-P" "/tmp/doctest/scripts/cmake/exec_test.cmake"
Directory: /tmp/doctest/build/examples/all_features
"test_cases_and_suites.cpp" start time: Oct 08 19:55 EDT
Output:
----------------------------------------------------------
-- Found Git: /usr/bin/git (found version "2.38.0") 
diff --git a/tmp/doctest/examples/all_features/test_output/test_cases_and_suites.cpp.txt b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_test_cases_and_suites.cpp.txt
index 299559d..7af8c19 100644
--- a/tmp/doctest/examples/all_features/test_output/test_cases_and_suites.cpp.txt
+++ b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_test_cases_and_suites.cpp.txt
@@ -40,7 +40,6 @@ TEST CASE:  normal test in a test suite from a decorator
 
 test_cases_and_suites.cpp(0): MESSAGE: failing because of the timeout decorator!
 
-Test case exceeded time limit of 0.000001!
 ===============================================================================
 test_cases_and_suites.cpp(0):
 DESCRIPTION: this test has overridden its skip decorator
@@ -95,7 +94,7 @@ test_cases_and_suites.cpp(0): ERROR:
 
 Didn't fail exactly 1 times so marking it as failed!
 ===============================================================================
-[doctest] test cases: 16 | 7 passed |  9 failed |
+[doctest] test cases: 16 | 8 passed |  8 failed |
 [doctest] assertions: 13 | 2 passed | 11 failed |
 [doctest] Status: FAILURE!
 Program code.
 
CMake Error at /tmp/doctest/scripts/cmake/exec_test.cmake:69 (message):
  Running '/tmp/doctest/build/examples/all_features/all_features
  --no-skipped-summary --no-version -sf=*test_cases_and_suites.cpp
  --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1
  --dt-gnu-file-line=0 --dt-no-time-in-output=1' ended with code 'Output is
  different from reference file!'
 
<end of output>
Test time =   0.12 sec
----------------------------------------------------------
Test Failed.
"test_cases_and_suites.cpp" end time: Oct 08 19:55 EDT
"test_cases_and_suites.cpp" time elapsed: 00:00:00
----------------------------------------------------------
```

That seems to have gotten rid of the timeout issue, but introduced three new failures (at least I think they are new).  Luckily these failures are deterministic at least.

```
98% tests passed, 3 tests failed out of 155

Total Test time (real) =  27.02 sec

The following tests FAILED:
         76 - coverage_maxout.cpp (Failed)
         77 - coverage_maxout.cpp_xml (Failed)
         78 - coverage_maxout.cpp_junit (Failed)
```

The output from the three failures is:

```
76/155 Testing: coverage_maxout.cpp
76/155 Test: coverage_maxout.cpp
Command: "/usr/bin/cmake" "-DCOMMAND=/tmp/doctest/build/examples/all_features/all_features --no-skipped-summary --no-version -sf=*coverage_maxout.cpp --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1" "-DTEST_OUTPUT_FILE=/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp.txt" "-DTEST_TEMP_FILE=/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp.txt" "-DTEST_MODE=COMPARE" "-P" "/tmp/doctest/scripts/cmake/exec_test.cmake"
Directory: /tmp/doctest/build/examples/all_features
"coverage_maxout.cpp" start time: Oct 08 23:15 EDT
Output:
----------------------------------------------------------
-- Found Git: /usr/bin/git (found version "2.38.0") 
diff --git a/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp.txt b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp.txt
index e0c193d..e69de29 100644
--- a/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp.txt
+++ b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp.txt
@@ -1,43 +0,0 @@
-[doctest] run with "--help" for options
-===============================================================================
-coverage_maxout.cpp(0):
-TEST CASE:  exercising tricky code paths of doctest
-
-coverage_maxout.cpp(0): ERROR: CHECK( str.compare(const_str, true) != 0 ) is NOT correct!
-  values: CHECK( 0 != 0 )
-  logged: should fail
-
-coverage_maxout.cpp(0): ERROR: CHECK( str.compare("omgomgomg", false) != 0 ) is NOT correct!
-  values: CHECK( 0 != 0 )
-  logged: should fail
-
-coverage_maxout.cpp(0): ERROR: CHECK_FALSE( isThereAnything ) is NOT correct!
-  values: CHECK_FALSE( true )
-  logged: should fail
-
-===============================================================================
-coverage_maxout.cpp(0):
-TEST SUITE: exception related
-TEST CASE:  will end from a std::string exception
-
-coverage_maxout.cpp(0): ERROR: test case THREW exception: std::string!
-
-===============================================================================
-coverage_maxout.cpp(0):
-TEST SUITE: exception related
-TEST CASE:  will end from a const char* exception
-
-coverage_maxout.cpp(0): ERROR: test case THREW exception: const char*!
-
-===============================================================================
-coverage_maxout.cpp(0):
-TEST SUITE: exception related
-TEST CASE:  will end from an unknown exception
-
-coverage_maxout.cpp(0): ERROR: test case THREW exception: unknown exception
-
-===============================================================================
-[doctest] test cases:  4 |  0 passed | 4 failed |
-[doctest] assertions: 31 | 28 passed | 3 failed |
-[doctest] Status: FAILURE!
-Program code.
 
CMake Error at /tmp/doctest/scripts/cmake/exec_test.cmake:69 (message):
  Running '/tmp/doctest/build/examples/all_features/all_features
  --no-skipped-summary --no-version -sf=*coverage_maxout.cpp
  --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1
  --dt-gnu-file-line=0 --dt-no-time-in-output=1' ended with code 'Output is
  different from reference file!'
 
<end of output>
Test time =   0.08 sec
----------------------------------------------------------
Test Failed.
"coverage_maxout.cpp" end time: Oct 08 23:15 EDT
"coverage_maxout.cpp" time elapsed: 00:00:00
----------------------------------------------------------

77/155 Testing: coverage_maxout.cpp_xml
77/155 Test: coverage_maxout.cpp_xml
Command: "/usr/bin/cmake" "-DCOMMAND=/tmp/doctest/build/examples/all_features/all_features --no-skipped-summary --no-version -sf=*coverage_maxout.cpp --reporters
=xml --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1" "-DTEST_OUTPUT_FILE=/tmp/doctest/examples/
all_features/test_output/coverage_maxout.cpp_xml.txt" "-DTEST_TEMP_FILE=/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp_
xml.txt" "-DTEST_MODE=COMPARE" "-P" "/tmp/doctest/scripts/cmake/exec_test.cmake"
Directory: /tmp/doctest/build/examples/all_features
"coverage_maxout.cpp_xml" start time: Oct 08 23:15 EDT
Output:
----------------------------------------------------------
-- Found Git: /usr/bin/git (found version "2.38.0") 
diff --git a/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp_xml.txt b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_cove
rage_maxout.cpp_xml.txt
index 9a9bf10..a50eb9d 100644
--- a/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp_xml.txt
+++ b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp_xml.txt
@@ -25,41 +25,3 @@
           should fail
         </Info>
       </Expression>
-      <Expression success="false" type="CHECK_FALSE" filename="coverage_maxout.cpp" line="0">
-        <Original>
-          isThereAnything
-        </Original>
-        <Expanded>
-          true
-        </Expanded>
-        <Info>
-          should fail
-        </Info>
-      </Expression>
-      <OverallResultsAsserts successes="28" failures="3" test_case_success="false"/>
-    </TestCase>
-  </TestSuite>
-  <TestSuite name="exception related">
-    <TestCase name="will end from a std::string exception" filename="coverage_maxout.cpp" line="0">
-      <Exception crash="false">
-        std::string!
-      </Exception>
-      <OverallResultsAsserts successes="0" failures="0" test_case_success="false"/>
-    </TestCase>
-    <TestCase name="will end from a const char* exception" filename="coverage_maxout.cpp" line="0">
-      <Exception crash="false">
-        const char*!
-      </Exception>
-      <OverallResultsAsserts successes="0" failures="0" test_case_success="false"/>
-    </TestCase>
-    <TestCase name="will end from an unknown exception" filename="coverage_maxout.cpp" line="0">
-      <Exception crash="false">
-        unknown exception
-      </Exception>
-      <OverallResultsAsserts successes="0" failures="0" test_case_success="false"/>
-    </TestCase>
-  </TestSuite>
-  <OverallResultsAsserts successes="28" failures="3"/>
-  <OverallResultsTestCases successes="0" failures="4"/>
-</doctest>
-Program code.

CMake Error at /tmp/doctest/scripts/cmake/exec_test.cmake:69 (message):
  Running '/tmp/doctest/build/examples/all_features/all_features
  --no-skipped-summary --no-version -sf=*coverage_maxout.cpp --reporters=xml
  --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1
  --dt-gnu-file-line=0 --dt-no-time-in-output=1' ended with code 'Output is
  different from reference file!'

<end of output>
Test time =   0.08 sec
----------------------------------------------------------
Test Failed.
"coverage_maxout.cpp_xml" end time: Oct 08 23:15 EDT
"coverage_maxout.cpp_xml" time elapsed: 00:00:00
----------------------------------------------------------

78/155 Testing: coverage_maxout.cpp_junit
78/155 Test: coverage_maxout.cpp_junit
Command: "/usr/bin/cmake" "-DCOMMAND=/tmp/doctest/build/examples/all_features/all_features --no-skipped-summary --no-version -sf=*coverage_maxout.cpp --reporters
=junit --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1" "-DTEST_OUTPUT_FILE=/tmp/doctest/example
s/all_features/test_output/coverage_maxout.cpp_junit.txt" "-DTEST_TEMP_FILE=/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp_junit.txt" "-DTEST_MODE=COMPARE" "-P" "/tmp/doctest/scripts/cmake/exec_test.cmake"
Directory: /tmp/doctest/build/examples/all_features
"coverage_maxout.cpp_junit" start time: Oct 08 23:15 EDT
Output:
----------------------------------------------------------
-- Found Git: /usr/bin/git (found version "2.38.0") 
diff --git a/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp_junit.txt b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp_junit.txt
index 8f1e992..e69de29 100644
--- a/tmp/doctest/examples/all_features/test_output/coverage_maxout.cpp_junit.txt
+++ b/tmp/doctest/build/examples/all_features/CMakeFiles/temp_test_output_coverage_maxout.cpp_junit.txt
@@ -1,44 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<testsuites>
-  <testsuite name="all_features" errors="3" failures="3" tests="31">
-    <testcase classname="coverage_maxout.cpp" name="exercising tricky code paths of doctest" status="run">
-      <failure message="0 != 0" type="CHECK">
-coverage_maxout.cpp(0):
-CHECK( str.compare(const_str, true) != 0 ) is NOT correct!
-  values: CHECK( 0 != 0 )
-  logged: should fail
-
-      </failure>
-      <failure message="0 != 0" type="CHECK">
-coverage_maxout.cpp(0):
-CHECK( str.compare("omgomgomg", false) != 0 ) is NOT correct!
-  values: CHECK( 0 != 0 )
-  logged: should fail
-
-      </failure>
-      <failure message="true" type="CHECK_FALSE">
-coverage_maxout.cpp(0):
-CHECK_FALSE( isThereAnything ) is NOT correct!
-  values: CHECK_FALSE( true )
-  logged: should fail
-
-      </failure>
-    </testcase>
-    <testcase classname="coverage_maxout.cpp" name="will end from a std::string exception" status="run">
-      <error message="exception">
-        std::string!
-      </error>
-    </testcase>
-    <testcase classname="coverage_maxout.cpp" name="will end from a const char* exception" status="run">
-      <error message="exception">
-        const char*!
-      </error>
-    </testcase>
-    <testcase classname="coverage_maxout.cpp" name="will end from an unknown exception" status="run">
-      <error message="exception">
-        unknown exception
-      </error>
-    </testcase>
-  </testsuite>
-</testsuites>
-Program code.
 
CMake Error at /tmp/doctest/scripts/cmake/exec_test.cmake:69 (message):
  Running '/tmp/doctest/build/examples/all_features/all_features
  --no-skipped-summary --no-version -sf=*coverage_maxout.cpp
  --reporters=junit --dt-no-path-filenames=1 --dt-no-line-numbers=1
  --dt-no-exitcode=1 --dt-gnu-file-line=0 --dt-no-time-in-output=1' ended
  with code 'Output is different from reference file!'
 
<end of output>
Test time =   0.08 sec
----------------------------------------------------------
Test Failed.
"coverage_maxout.cpp_junit" end time: Oct 08 23:15 EDT
"coverage_maxout.cpp_junit" time elapsed: 00:00:00
----------------------------------------------------------
```

--- a/examples/all_features/test_cases_and_suites.cpp
+++ b/examples/all_features/test_cases_and_suites.cpp
@@ -2,6 +2,11 @@
 
 #include "header.h"
 
+DOCTEST_MAKE_STD_HEADERS_CLEAN_FROM_WARNINGS_ON_WALL_BEGIN
+#include <chrono>
+#include <thread>
+DOCTEST_MAKE_STD_HEADERS_CLEAN_FROM_WARNINGS_ON_WALL_END
+
 static int doStuff() {
     int a = 5;
     a += 2;
@@ -46,6 +51,14 @@ TEST_CASE_FIXTURE(SomeFixture, "fixtured test - not part of a test suite") {
 
 TEST_CASE("normal test in a test suite from a decorator" * doctest::test_suite("ts1") *
           doctest::timeout(0.000001)) {
+    // Guarantee that we time-out by sleeping for at least 5ms (> 1us)
+    using clock_type = std::chrono::steady_clock;
+    const auto quanta = std::chrono::milliseconds(5);
+    const auto t0 = clock_type::now();
+    while (clock_type::now() < t0 + quanta) {
+        std::this_thread::sleep_for(quanta);
+    }
+
     MESSAGE("failing because of the timeout decorator!");
 }
 

